spec/factories/service_providers.rb

```diff
@@ -1,7 +1,7 @@
 FactoryGirl.define do
   factory :service_provider do
     sequence(:friendly_name) { |n| "test-service_provider-#{n}" }
-    sequence(:issuer) { |n| "test-service_provider-#{n}" }
+    sequence(:issuer) { |n| "urn:gov:gsa:SAML:2.0.profiles:sp:sso:DEPT:APP-#{n}" }
```

Comment:
Are you thinking about changing the `issuer` to `issuer_department` and `issuer_agency` in the model? I'm not sure that will work with the ServiceProviders that have been created with an issuer that does not match the format.

If we want to keep the schema that way it is and only change it in the UI and build the issuer in the controller, then the challenge is:

1. How to actually lay this out in the UI. We'd want a control that communicates that the `DEPT` and `APP` strings are components of a uid.
2. Parsing the `DEPT` and `APP` strings out of the `issuer`, especially for existing service providers who have what we'd now describe as a malformed issuer.